fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/F-PLN.tsx

```diff
@@ -0,0 +1,1068 @@
+                                action: () => this.props.navigateTo(`fms/${this.props.activeUri.get().category}/f-pln-alternate`),
+                            },
+                            {
+                                label: 'CLOSEST AIRPORTS',
+                                action: () => this.props.navigateTo(`fms/${this.props.activeUri.get().category}/f-pln-closest-airports`),
+                            },
+                            {
+                                label: 'EQUI-TIME POINT',
+                                action: () => this.props.navigateTo(`fms/${this.props.activeUri.get().category}/f-pln-equi-time-point`),
+                            },
+                            {
+                                label: 'FIX INFO',
+                                action: () => this.props.navigateTo(`fms/${this.props.activeUri.get().category}/f-pln-fix-info`),
+                            },
+                            {
+                                label: 'LL CROSSING',
+                                action: () => this.props.navigateTo(`fms/${this.props.activeUri.get().category}/f-pln-ll-xing-time-mkr`),
+                            },
+                            {
+                                label: 'TIME MARKER',
+                                action: () => this.props.navigateTo(`fms/${this.props.activeUri.get().category}/f-pln-ll-xing-time-mkr`),
+                            },
+                            {
+                                label: 'CPNY F-PLN REPORT',
+                                action: () => null,
+                            }])}
+                        />
+                        <Button label="DIR TO" onClick={() => this.props.navigateTo(`fms/${this.props.activeUri.get().category}/f-pln-direct-to`)} buttonStyle="margin-right: 5px;" />
+                    </div>
+                    {/* end page content */}
+                </div>
+                <Footer bus={this.props.bus} activeUri={this.props.activeUri} navigateTo={this.props.navigateTo} />
+            </>
+        );
+    }
+}
+
+interface FplnLineCommonProps extends ComponentProps {
+    openRevisionsMenuCallback: () => void;
+}
+
+enum FplnLineType {
+    Waypoint,
+    Special,
+  }
+
+enum FplnLineColor {
+    Active = '#00ff00',
+    Temporary = '#ffff00',
+    Secondary = '#ffffff'
+  }
+
+enum FplnLineFlags {
+    None = 0,
+    FirstLine = 1 << 0,
+    BeforeSpecial = 1 << 1,
+    AfterSpecial = 1 << 2,
+    BeforeActiveLeg = 1 << 3,
+    IsActiveLeg = 1 << 4,
+    LastLine = 1 << 5,
+  }
+interface FplnLineTypeDiscriminator {
+    /*
+     * waypoint: Regular or pseudo waypoints
+     * special: DISCONTINUITY, END OF F-PLN etc.
+     */
+    type: FplnLineType;
+    originalLegIndex: number; // TODO consider linking to flight plan when data is being integrated
+}
+
+// Type for DISCO, END OF F-PLN etc.
+interface FplnLineSpecialDisplayData extends FplnLineTypeDiscriminator {
+    label: string;
```

Comment:
Then I would get the following error at line 759 (const delimiter = data.label.length > 13 ? '- - - - -' : '- - - - - - ';):
Property 'label' does not exist on type 'never'.\n  The intersection 'FplnLineSpecialDisplayData & FplnLineWaypointDisplayData' was reduced to 'never' because property 'type' has conflicting types in some constituents

Reply:
This just seems stupidly complicated in TS: https://stackoverflow.com/questions/72877582/discriminated-type-unions-in-typescript-with-lookup-object